### PR TITLE
allow h2 and webrtc to move forward concurrently

### PR DESCRIPTION
**Pull request by npmenard**
This slightly change the connection flow allowing both webrtc & http2 connection to move forward concurrently. This will also give us the option to increase the number of concurrent answerer. Note that there can only be one answerer listening on the /Answer endpoint at any time.

### CODE REVIEW

**Review by acmorrow — APPROVED** [review]
LGTM per discussion

File: micro-rdk/src/common/conn/viam.rs [hunk 1]
```diff
@@ -865,36 +881,34 @@ where
...
+        let sdp = sig.wait_sdp_offer().await?;
+        Ok((sig, sdp))
     }
 }
 
 struct SignalingTask {
-    sender: Sender<AppSignaling>,
+    sender: Sender<Box<WebRtcSignalingChannel>>,
     rpc_host: String,
+    max_answerer: Arc<async_lock::Semaphore>,
```
Comment (gvaradarajan) [inline micro-rdk/src/common/conn/viam.rs:911]: fancy

**Review by gvaradarajan — APPROVED** [review]
LGTM

**Review by mattjperez — APPROVED** [review]
LGTM

**Pull request merged**